components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAOImpl.java

```diff
@@ -0,0 +1,328 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2.device.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.device.constants.Constants;
+import org.wso2.carbon.identity.oauth2.device.model.DeviceFlowDO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+import java.util.UUID;
+
+/**
+ * This class contains override methods of DeviceFlowDAO.
+ */
+public class DeviceFlowDAOImpl implements DeviceFlowDAO {
+
+    private static final Log log = LogFactory.getLog(DeviceFlowDAOImpl.class);
+
+    private String clientId;
+    private String status;
+    private String scope;
+
+    @Override
+    public void insertDeviceFlow(String deviceCode, String userCode, String consumerKey, String scope, Long expiresIn,
+                                 int interval) throws IdentityOAuth2Exception {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLQueries.DeviceFlowDAOSQLQueries.STORE_DEVICE_CODE)) {
+                Date date = new Date();
+                Timestamp timeCreated = new Timestamp(date.getTime());
+                long timeExpired = timeCreated.getTime() + expiresIn;
+                prepStmt.setString(1, UUID.randomUUID().toString());
+                prepStmt.setString(2, deviceCode);
+                prepStmt.setString(3, userCode);
+                prepStmt.setString(4, scope);
+                prepStmt.setTimestamp(5, timeCreated, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setTimestamp(6, timeCreated, Calendar.getInstance(TimeZone
+                        .getTimeZone(Constants.UTC)));
+                prepStmt.setLong(7, timeExpired);
+                prepStmt.setLong(8, interval);
+                prepStmt.setString(9, Constants.PENDING);
+                prepStmt.setString(10, consumerKey);
+                prepStmt.execute();
+                IdentityDatabaseUtil.commitTransaction(connection);
+            } catch (SQLException e) {
```

Comment:
Is this catch block needed?

Reply:
Yes, it is needed. We need to roll back the transaction. But since connection declared in the try block, can't use the same catch block.